### PR TITLE
feature: add github actions CI; optimized dockerfile for docker cache

### PR DESCRIPTION
**Pull request by maze88**
- utilizes existing makefile for github actions CI workflow.
- swapped order of COPY and LABEL layers in dockerfile, to improve docker image cache utilization.
- remove unused legacy jenkinsfile.

### CODE REVIEW

**Review by e0ne — COMMENTED** [review]
LGTM. Please, split changes to Dockerfile and GitHub Actions into different commits

**Comment by maze88** [comment]
> LGTM. Please, split changes to Dockerfile and GitHub Actions into different commits

Thanks Ivan. Splitted.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Multi-architecture Docker Build
```
[the remark this answers is not in the document]
Comment (maze88) [reply .github/workflows/ci.yaml:1]: Would seeing [a test run from my fork](https://github.com/maze88/network-operator/actions/runs/8907620626) be sufficient?

And I've changed workflow filename, as recommended.

[the remark this answers is not in the document]
Comment (killianmuldoon) [reply .github/workflows/ci.yaml]: Thanks - could you put that on your main branch and run it so we can ensure the `latest` handling is as expected?

[the remark this answers is not in the document]
Comment (maze88) [reply .github/workflows/ci.yaml]: Already [tested with alternate branch name](https://github.com/maze88/network-operator/actions/runs/8902558849/job/24448695776#step:3:5)... the [result](https://github.com/maze88/network-operator/actions/runs/8902558849/job/24448695776#step:5:13) is as expected:
`Docker tags will be: 9fe0355 latest`

**Review by killianmuldoon — COMMENTED** [review]
Looks good to me - should be good to merge once CI passes.

Thanks for working on this!

**Review by almaslennikov — APPROVED** [review]

**Review by rollandf — APPROVED** [review]

**Pull request merged**